compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -344,10 +344,16 @@ impl DataFlowGraph {
     pub(crate) fn get_value_max_num_bits(&self, value: ValueId) -> u32 {
         match self[value] {
             Value::Instruction { instruction, .. } => {
+                let bit_size = self.type_of_value(value).bit_size();
                 if let Instruction::Cast(original_value, _) = self[instruction] {
-                    self.type_of_value(original_value).bit_size()
+                    let original_bit_size = self.type_of_value(original_value).bit_size();
+                    if original_bit_size < bit_size {
+                        original_bit_size
+                    } else {
+                        bit_size
+                    }
```

Comment:
This method is used for determining whether we need a range check. So if a value was previously cast it is bounded by the smaller bit size. This is at least what this logic is trying to do, so I am just trying to prevent us from actually moving to a larger bound.

Reply:
If this cast logic was not here we would just have this:
```
Value::Instruction { instruction, .. } => {
     self.type_of_value(value).bit_size()
}
```